config/routes.rb

```diff
@@ -13,20 +13,25 @@
 
   resources :suppressions, except: [:show, :edit, :update, :destroy]
 
+  # Push routes
   namespace 'jira' do
     namespace 'status' do
+      # These require :service_name as a query string param
       resources :push, only: [:edit, :update]
     end
   end
 
-  get 'sha/:id' => 'jira/status/push#edit'
-  get 'deploy_email/:id' => 'jira/status/push#deploy_email'
-  get 'branch/:branch' => 'jira/status/push#branch'
-  get 'summary' => 'jira/status/push#summary'
+  get  ':service_name/sha/:id'          => 'jira/status/push#edit'
+  get  ':service_name/deploy_email/:id' => 'jira/status/push#deploy_email'
+  get  ':service_name/summary'          => 'jira/status/push#summary'
```

Comment:
Given that there may be bookmarks to the original `'summary'` route, for compatibility it'd probably be good to keep a `'summary'` route that re-routes to the summary for `web`. What do you think?

Reply:
that's a good point! Rather than keeping two `summary` routes that are nearly identical, which would be confusing for future devs, I would rather keep only one route and expect `service_name` as a query param, so it's easier to implement a default value of `"web"`